### PR TITLE
IOZone Test for avocado

### PR DESCRIPTION
**Pull request by harish-24**
Ported IOZone test from autotest-client-tests to avocado-misc-tests

### CODE REVIEW

File: io/disk/iozone.py [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+_LABELS = ['file_size', 'record_size', 'write', 'rewrite', 'read', 'reread',
+           'randread', 'randwrite', 'bkwdread', 'recordrewrite', 'strideread',
+           'fwrite', 'frewrite', 'fread', 'freread']
+
+
+def unique(list):
```
Comment (ldoktor) [inline io/disk/iozone.py:21]: It's not really nice to replace builtins with params, please use something else than list. (eg. items, or inp_list)

Comment (ldoktor) [inline io/disk/iozone.py:21]: Well actually we do have similar function, even ordered in `avocado.utils.data_structures` called `ordered_list_unique`. I think it'd better to use it here. (it's opimized)

File: io/disk/iozone.py [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+        return u.keys()
+
+
+def geometric_mean(values):
+    """
+    Evaluates the geometric mean for a list of numeric values.
+
+    :param values: List with values.
+    :return: Single value representing the geometric mean for the list values.
+    @see: http://en.wikipedia.org/wiki/Geometric_mean
```
Comment (ldoktor) [inline io/disk/iozone.py:47]: This is the `docstring` notation, would you please use the `sphinx` one instead (`:see: ...`)

Comment (ldoktor) [inline io/disk/iozone.py:47]: Actually this is a good candidate for `avocado.utils.data_structures`. Could you please send a PR there and use it from there?

File: io/disk/iozone.py [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+        for element1, element2 in zip(line1, line2):
+            ratio = float(element2) / float(element1)
+            if ratio < (1 - treshold):
+                regressions += 1
+                new_line.append((100 * ratio - 1) - 100)
+            elif ratio > (1 + treshold):
+                improvements += 1
+                new_line.append("+" + str((100 * ratio - 1) - 100))
+            else:
+                same + 1
```
Comment (ldoktor) [inline io/disk/iozone.py:86]: There is probably a missing `=`

File: io/disk/iozone.py [hunk 4]
```diff
@@ -0,0 +1,649 @@
+import os
```
Comment (ldoktor) [inline io/disk/iozone.py:1]: As you port this from autotest, you should mention the original owner and the same/compatible license on top of this file. (see https://github.com/avocado-framework/avocado/blob/master/avocado/utils/software_manager.py as example)

Comment (ldoktor) [inline io/disk/iozone.py:1]: PS: I noticed we forgot doing this in all other tests, I'll take care of them and I'd like you to take care of this one.

File: io/disk/iozone.py [hunk 5]
```diff
@@ -0,0 +1,649 @@
...
+    comparison_matrix = []
+
+    new_matrix = []
+    for line1, line2 in zip(matrix1, matrix2):
+        new_line = []
+        for element1, element2 in zip(line1, line2):
+            ratio = float(element2) / float(element1)
+            if ratio < (1 - treshold):
+                regressions += 1
+                new_line.append((100 * ratio - 1) - 100)
```
Comment (ldoktor) [inline io/disk/iozone.py:81]: the parenthesis around `(100 * ratio - 1)` are not needed.

File: io/disk/iozone.py [hunk 6]
```diff
@@ -0,0 +1,649 @@
...
+            elif ratio > (1 + treshold):
+                improvements += 1
+                new_line.append("+" + str((100 * ratio - 1) - 100))
```
Comment (ldoktor) [inline io/disk/iozone.py:84]: the same here

File: io/disk/iozone.py (hunk 5 above)
Comment (ldoktor) [inline io/disk/iozone.py:81]: Actually why is there the `-1`? It should be only `100 * ratio - 100` in order to get the percentage increase/decrease

File: io/disk/iozone.py [hunk 7]
```diff
@@ -0,0 +1,649 @@
...
+            else:
+                same + 1
+                if line1.index(element1) == 0:
```
Comment (ldoktor) [inline io/disk/iozone.py:87]: I don't understand the purpose of this. Basically if the value of `element1` is equal to the first value of this line, it adds the `element1` value, otherwise it adds `.`.

I think it should only add `.` (to emphasize the value has not changed), or always the value (without the `+|-` so people know it hasn't changed), what do you think?

File: io/disk/iozone.py [hunk 8]
```diff
@@ -0,0 +1,649 @@
...
+    improvements = 0
+    regressions = 0
+    same = 0
+    comparison_matrix = []
+
+    new_matrix = []
+    for line1, line2 in zip(matrix1, matrix2):
+        new_line = []
+        for element1, element2 in zip(line1, line2):
+            ratio = float(element2) / float(element1)
```
Comment (ldoktor) [inline io/disk/iozone.py:78]: This is not `element1 == 0` safe.

File: io/disk/iozone.py [hunk 9]
```diff
@@ -0,0 +1,649 @@
...
+    except ValueError:
+        return None
+    product = 1
+    n = len(values)
+    if n == 0:
+        return None
+    return math.exp(sum([math.log(x) for x in values]) / n)
+
+
+def compare_matrices(matrix1, matrix2, treshold=0.05):
```
Comment (ldoktor) [inline io/disk/iozone.py:60]: Actually this also looks quite useful for the `avocado.utils.data_structures`, what do you think?

File: io/disk/iozone.py [hunk 10]
```diff
@@ -0,0 +1,649 @@
...
+class IOzoneAnalyzer(object):
+
+    """
+    Postprocessing module for IOzone. It is capable to pick results from an
+    IOzone run, calculate the geometric mean for all throughput results for
+    a given file size or record size, and then generate a series of 2D and 3D
+    graphs. The graph generation functionality depends on gnuplot, and if it
+    is not present, functionality degrates gracefully.
+
+    @copyright: Red Hat 2010
```
Comment (ldoktor) [inline io/disk/iozone.py:107]: First please use `:copyright:` instead and then are you sure with this copyright? In Autotest I see `@author: Ying Tao` from IBM.

Comment (ldoktor) [inline io/disk/iozone.py:107]: Anyway it's not suppose to be here, it should be on top of the file, see the `software_manager.py` example.

File: io/disk/iozone.py [hunk 11]
```diff
@@ -0,0 +1,649 @@
...
+        logging.info("TABLE:  RECsize Difference between runs                 "
+                     "Results are % DIFF")
+        logging.info("")
+        logging.info("RECORD    INIT    RE              RE    RANDOM  RANDOM  "
+                     "BACKWD   RECRE  STRIDE    F       FRE     F       FRE ")
+        logging.info("SIZE (KB) WRITE   WRITE   READ    READ    READ   WRITE  "
+                     "READ   WRITE    READ    WRITE   WRITE   READ    READ")
+        logging.info("-------------------------------------------------------"
+                     "-------------------------------------------------------")
+        for result_line in record_size:
```
Comment (ldoktor) [inline io/disk/iozone.py:303]: How about using `tabular_output` from `avocado/utils/astring.py`?

File: io/disk/iozone.py [hunk 12]
```diff
@@ -0,0 +1,649 @@
...
+                     (100 * record_regressions / float(record_total)),
+                     record_improvements,
+                     (100 * record_improvements / float(record_total)))
+        logging.info("")
+
+        logging.info("")
+        logging.info("TABLE:  FILEsize Difference between runs               "
+                     "Results are % DIFF")
+        logging.info("")
+        logging.info("RECORD    INIT    RE              RE    RANDOM  RANDOM "
```
Comment (ldoktor) [inline io/disk/iozone.py:318]: The same here, the tabular_output is much nicer (and safer for bigger numbers)

File: io/disk/iozone.py [hunk 13]
```diff
@@ -0,0 +1,649 @@
...
+        tarball = self.fetch_asset(
+            'http://www.iozone.org/src/current/iozone3_434.tar')
+        archive.extract(tarball, self.srcdir)
+        version = os.path.basename(tarball.split('.tar')[0])
+        self.srcdir = os.path.join(self.srcdir, version)
+
+        make_dir = os.path.join(self.srcdir, 'src/current')
+        os.chdir(make_dir)
+        patch = self.params.get('patch', default='makefile.patch')
+        process.run('patch -p3 < %s' % data_dir + '/' + patch, shell=True)
```
Comment (ldoktor) [inline io/disk/iozone.py:515]: please use `os.path.join()` instead of `+ '/' +`

File: io/disk/iozone.py [hunk 14]
```diff
@@ -0,0 +1,649 @@
...
+        data_dir = os.path.abspath(self.datadir)
+        self.base_dir = os.path.abspath(self.basedir)
+
+        tarball = self.fetch_asset(
+            'http://www.iozone.org/src/current/iozone3_434.tar')
+        archive.extract(tarball, self.srcdir)
+        version = os.path.basename(tarball.split('.tar')[0])
+        self.srcdir = os.path.join(self.srcdir, version)
+
+        make_dir = os.path.join(self.srcdir, 'src/current')
```
Comment (ldoktor) [inline io/disk/iozone.py:512]: here too (`os.path.join()`)

File: io/disk/iozone.py [hunk 15]
```diff
@@ -0,0 +1,649 @@
...
+                        if match:
+                            result = match.group(1)
+                            key_name = "%s-%d-%s" % (section, w_count, basekey)
+                            keylist[key_name] = result
+        output_path = os.path.join(self.outputdir, "perf.json")
+        json.dump(keylist, open(output_path, "w"), indent=1)
+
+    def test(self):
+
+        dir = self.params.get('dir', default=None)
```
Comment (ldoktor) [inline io/disk/iozone.py:613]: Please do no override builtins

File: io/disk/iozone.py.data/iozone.yaml [hunk 16]
```diff
@@ -0,0 +1,7 @@
+setup:
+    location: !mux
+        default:
+            dir: null
```
Comment (ldoktor) [inline io/disk/iozone.py.data/iozone.yaml:4]: Would you please describe as comments the `dir` and `args`?

File: io/disk/iozone.py [hunk 17]
```diff
@@ -0,0 +1,649 @@
...
+
+    :param values: List with values.
+    :return: Single value representing the geometric mean for the list values.
+    @see: http://en.wikipedia.org/wiki/Geometric_mean
+    """
+    try:
+        values = [int(value) for value in values]
+    except ValueError:
+        return None
+    product = 1
```
Comment (ldoktor) [inline io/disk/iozone.py:53]: This variable is not used.

File: io/disk/iozone.py [hunk 18]
```diff
@@ -0,0 +1,649 @@
...
+
+    :param matrix1: Reference Matrix with numeric data
+    :param matrix2: Matrix that will be compared
+    :param treshold: Any difference bigger than this percent treshold will be
+            reported.
+    """
+    improvements = 0
+    regressions = 0
+    same = 0
+    comparison_matrix = []
```
Comment (ldoktor) [inline io/disk/iozone.py:72]: This variable is not used

**Comment by ldoktor** [comment]
@harish-24 might I ask you to run `pylint io/disk/iozone.py` and would you please take care of all errors, most of warnings and ideally some of the conventions?

File: io/disk/iozone.py [hunk 19]
```diff
@@ -0,0 +1,649 @@
...
+        f = open(self.results_path, 'w')
+        try:
+            f.write(self.results)
+        finally:
+            f.close()
+
+        self.generate_keyval()
+
+        if self.auto_mode:
+            a = IOzoneAnalyzer(list_files=[self.results_path],
```
Comment (ldoktor) [inline io/disk/iozone.py:640]: OK so the comparison is actually never used as you only support single `list_files` file. How about allowing `params.get(previous_results, None)` and if not None append it here so people might actually use it?

I tried it manually by providing previous data results and it seems to be working:

```
ANALYSIS of DRILLED DATA:

TABLE:  RECsize Difference between runs                 Results are % DIFF

RECORD    INIT    RE              RE    RANDOM  RANDOM  BACKWD   RECRE  STRIDE    F       FRE     F       FRE 
SIZE (KB) WRITE   WRITE   READ    READ    READ   WRITE  READ   WRITE    READ    WRITE   WRITE   READ    READ
--------------------------------------------------------------------------------------------------------------
4         +6.262  .       .       -8.366  -9.828  -10.40  +6.320  .       -6.743  -6.660  .       -12.33  .       
8         .       -11.74  .       -21.41  -9.530  .       -17.19  .       -20.49  .       -8.594  .       +8.815  
16        -8.796  .       +15.52  +5.163  +12.70  +6.207  +8.601  .       -16.18  -15.68  -11.26  .       +12.22  
32        .       .       -15.59  -18.29  .       -12.12  -18.49  -8.060  -20.69  -6.581  .       -9.863  -15.73  
64        .       .       .       -7.431  -9.286  .       -9.756  .       -10.61  .       .       .       -7.233  
128       .       .       .       .       -6.704  .       -11.20  .       .       .       -6.741  .       -12.61  
256       -8.730  .       .       .       .       .       +4.246  .       .       .       .       .       .       
512       .       +4.160  .       -9.914  -12.89  -6.136  .       -6.050  -8.972  -7.337  .       .       -10.27  
1024      .       .       .       .       .       .       -7.605  .       -12.47  .       .       .       -7.058  
2048      .       .       .       -8.330  -8.480  .       -6.871  -7.931  .       .       .       .       .       
4096      .       .       .       +7.829  +7.992  +9.787  +4.638  +13.42  .       .       .       .       .       
8192      .       .       .       .       .       +7.965  +5.080  +8.345  +5.943  .       .       -11.26  +7.264  
16384     -8.722  .       .       .       .       .       .       .       .       .       -8.733  .       .       
REGRESSIONS: 52 (28.57%)    Improvements: 21 (11.54%)


TABLE:  FILEsize Difference between runs               Results are % DIFF

RECORD    INIT    RE              RE    RANDOM  RANDOM BACKWD   RECRE  STRIDE    F       FRE     F       FRE 
SIZE (KB) WRITE   WRITE   READ    READ    READ   WRITE READ   WRITE    READ    WRITE   WRITE   READ    READ
--------------------------------------------------------------------------------------------------------------
64        .       -9.863  .       -16.26  -7.558  -12.22  +11.23  +12.43  -25.95  .       -13.02  .       +49.07  
128       +10.42  .       .       -13.60  -11.65  .       -28.90  -26.85  -11.80  -12.69  +4.483  -6.698  +29.96  
256       .       -13.05  .       .       .       -11.82  +18.94  +7.166  -9.394  .       +8.011  .       -7.661  
512       .       .       -8.629  .       +6.799  +4.632  -17.58  -6.751  -6.571  .       -15.49  .       -26.72  
1024      -8.580  .       .       -19.98  .       +4.641  -18.82  .       -19.62  .       .       .       -20.35  
2048      .       +7.071  -6.958  -17.84  -20.31  .       .       .       -9.327  .       .       -6.058  -16.23  
4096      .       .       -6.227  .       -15.18  -6.789  .       +4.406  -17.42  .       .       .       .       
8192      .       .       .       +4.258  +4.722  .       .       .       -10.76  .       .       .       .       
16384     .       .       -6.143  -6.089  .       .       .       .       .       .       .       .       .       
32768     +5.035  .       +4.427  +4.924  .       +6.517  .       .       .       .       .       .       .       
65536     .       .       .       .       .       +4.154  .       .       +7.563  .       .       .       .       
131072    .       .       .       .       .       +4.115  .       .       .       .       .       .       .       
262144    -9.357  .       -9.898  -6.150  .       .       -15.17  -7.130  -14.80  -6.117  -9.779  -10.18  .       
524288    -8.684  .       .       -7.075  -9.334  .       .       -7.676  .       .       .       -6.640  .       
REGRESSIONS: 55 (28.06%)    Improvements: 23 (11.73%)
```

File: io/disk/iozone.py (hunk 7 above)
Comment (ldoktor) [inline io/disk/iozone.py:87]: Actually after using this function I understand it. The first element is the header. So how about using `enumerate` and `if i != 0`?

**Comment by ldoktor** [comment]
@harish-24 I noticed there are many failures while plotting the results:

```
...
20:02:54 DEBUG| [stderr] "/home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/write.do", line 18: warning: No usable data in this plot to auto-scale axis range
20:02:54 DEBUG| [stderr]
20:02:54 DEBUG| [stderr] splot '/home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/3d-datasource' using 1:2:1 title 'write'
20:02:54 DEBUG| [stderr]                                                                                                                                                                          ^
20:02:54 DEBUG| [stderr] "/home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/write.do", line 18: All points z value undefined
20:02:54 DEBUG| [stderr]
20:02:54 INFO | Running 'gnuplot /home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/rewrite.do'
20:02:54 DEBUG| [stderr] "/home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/rewrite.do", line 18: warning: No usable data in this plot to auto-scale axis range
20:02:54 DEBUG| [stderr]
20:02:54 DEBUG| [stderr] splot '/home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/3d-datasource' using 1:2:2 title 'rewrite'
20:02:54 DEBUG| [stderr]                                                                                                                                                                            ^
20:02:54 DEBUG| [stderr] "/home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/rewrite.do", line 18: All points z value undefined
20:02:54 DEBUG| [stderr]
20:02:54 INFO | Running 'gnuplot /home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/read.do'
20:02:54 DEBUG| [stderr] "/home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/read.do", line 18: warning: No usable data in this plot to auto-scale axis range
20:02:54 DEBUG| [stderr]
20:02:54 DEBUG| [stderr] splot '/home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/3d-datasource' using 1:2:3 title 'read'
20:02:54 DEBUG| [stderr]                                                                                                                                                                         ^
20:02:54 DEBUG| [stderr] "/home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/read.do", line 18: All points z value undefined
20:02:54 DEBUG| [stderr]
20:02:54 INFO | Running 'gnuplot /home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/reread.do'
20:02:54 DEBUG| [stderr] "/home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/reread.do", line 18: warning: No usable data in this plot to auto-scale axis range
20:02:54 DEBUG| [stderr]
20:02:54 DEBUG| [stderr] splot '/home/medic/avocado/job-results/job-2016-05-17T20.00-4af297f/test-results/1-io_disk_iozone.py:IOZone.test/data/analysis_1/3d-datasource' using 1:2:4 title 'reread'
20:02:54 DEBUG| [stderr]                                                                                                                                                                           ^
...
```

Some of the graphs are plotted, but most of them are empty. Can you please address those issues?

File: io/disk/iozone.py (hunk 19 above)
Comment (ldoktor) [inline io/disk/iozone.py:640]: Actually you might skip this one as it's an improvement and not port. Or if you're interested, you can include it as a separate commit.

File: io/disk/iozone.py (hunk 4 above)
Comment (ldoktor) [inline io/disk/iozone.py:1]: Actually you can rather inspire here: https://github.com/avocado-framework/avocado-misc-tests/pull/26 for heading (both versions are acceptable)

File: io/disk/iozone.py (hunk 1 above)
Comment (harish-24) [inline io/disk/iozone.py:21]: ordered_list_unique can be used here. Thanks.

File: io/disk/iozone.py (hunk 2 above)
Comment (harish-24) [inline io/disk/iozone.py:47]: Will change `docstring` notations.  
`geometric_mean` can be added to `avocado.utils.data_structures`.

File: io/disk/iozone.py (hunk 9 above)
Comment (harish-24) [inline io/disk/iozone.py:60]: Sure. Will send a separate PR in avocado for compare_matrices.

File: io/disk/iozone.py (hunk 4 above)
Comment (harish-24) [inline io/disk/iozone.py:1]: Will follow the same as #26

**Comment by harish-24** [comment]
@ldoktor  The problem persists even in autotest for plotting 3d graphs.Can this be an improvement?

File: io/disk/iozone.py.data/iozone.yaml (hunk 16 above)
Comment (harish-24) [inline io/disk/iozone.py.data/iozone.yaml:4]: Sure. Can add these.

**Comment by ldoktor** [comment]
Sure, I guess no one needs them right now. (3d graphs)

**Comment by lmr** [comment]
@harish-24 and @ldoktor: Basically you guys are discussing code I wrote years ago. I'm happy to see that ported over to avocado, but please do acknowledge the original author (me).

**Comment by lmr** [comment]
About the problem plotting graphs, it's simple: The original code was written for the gnuplot version I currently had installed on my laptop while developing. Time passed, changes were made to gnuplot and now the syntax used here is incompatible. So some update is due here.

**Comment by lmr** [comment]
TBH, an even better solution would be to remove the need for the external utility gnuplot and use matplotlib to plot up the graphs.

File: io/disk/iozone.py (hunk 10 above)
Comment (lmr) [inline io/disk/iozone.py:107]: @ldoktor Oh, and the original author of the iozone wrapper was Ying Tao, but all of the analysis added later was my work and the copyright is Red Hat, since I worked for Red Hat at the time.

File: io/disk/iozone.py (hunk 17 above)
Comment (lmr) [inline io/disk/iozone.py:53]: I meant to do product = math.exp(sum([math.log(x) for x in values]) / n) and then return product, then forgot about it halfway, I'd guess.

File: io/disk/iozone.py (hunk 19 above)
Comment (lmr) [inline io/disk/iozone.py:640]: This comparison script was meant to be used with the results of 2 runs, so there was no way of using it directly at test runtime.

**Comment by ldoktor** [comment]
Nice, I didn't know. There is no license in autotest and the original commit is from:

```
commit a59abddc45a8bc9e9074a2ef1bde24ba823eb694
Author: Ying Tao <yingtao@cn.ibm.com>
Date:   Thu Oct 19 14:36:45 2006 +0000

    Add iozone test

    From: Ying Tao <yingtao@cn.ibm.com>
```

I see your (extensive) contributions, but you never modified the copyright, so I think Ying Tao is the one who should be listed. (even though his version was quite simple).

About the matplotlib, yes, it'd be nice, but porting the code is the first step. When a user needs the modification, they should suggest or change it.

**Comment by ldoktor** [comment]
But I'm not a lawyer. I only heard that according to US legislation the author is the first one, or any contributor with significant changes if he claims them (using (C)). According to German legislation, the authorship fades and as you modified most of the code, you'd be probably the author. GPL says it's the first committer, or anyone who adds (c) along with significant modification (so the USA way).

So if you insist, I think @harish-24 could add both of you, but you (@lmr) should mention the years you worked on it as it's not written anywhere else. Then it might look probably like this:

```
# Based on code by Ying Tao <yingtao@cn.ibm.com>
#   copyright: 2006 IBM
#   copyright: xxx - yyy Lucas Meneghel Rodrigues <lmr@redhat.com>
```

But I'm not sure, because I think the copyright belongs to Red Hat (correct me if I'm wrong), so how about:

```
# Based on code by:
#   author: Ying Tao <yingtao@cn.ibm.com>
#   author: Lucas Meneghel Rodrigues <lmr@redhat.com>
#   copyright: 2006 IBM
#   copyright: xxx - yyy Red Hat, Inc.
```

File: io/disk/iozone.py (hunk 7 above)
Comment (harish-24) [inline io/disk/iozone.py:87]: @ldoktor In `compare_matrices`, how would you suggest to handle the headers
(RECORD SIZE (KB) 4,8,16.. in this case) of the input files ? Can we exclude the headers using `enumerate` and `if i != 0` ? Or Is it okay to append the header as it is in the list ?

Comment (ldoktor) [inline io/disk/iozone.py:87]: I'd suggest to pop out the first element and then enter the for loop, something like:

``` python
from itertools import izip
# ...
newline = []
elements = izip(line1, line2)
try:
    element1, element2 = elements.next()
except StopIteration:    # there are no elements in this line
    new_matrix.append(newline)
    continue
if element1 == element2:    # It's header
    newline.append(element1)
    try:
        element1, element2 = elements.next()
    except StopIteration:    # Only header in this line
        new_matrix.append(newline)
        continue
while True:
    # body of the for loop
    try:
        element1, element2 = elements.next()
    except StopIteration:
        break
```

File: io/disk/iozone.py (hunk 19 above)
Comment (ldoktor) [inline io/disk/iozone.py:640]: Cool, good to know.

File: io/disk/iozone.py (hunk 8 above)
Comment (harish-24) [inline io/disk/iozone.py:78]: @ldoktor The best thing that can be done here is to catch the exception and display a message. Do you suggest to do that?

Comment (ldoktor) [inline io/disk/iozone.py:78]: Well we can handle that without the division, which would add an overhead and as this is a corner case, I'd suggest using
try:
    ratio = float(element2) / float(element1)
except ZeroDivisionError:
    new_line.append("." if float(element2) == 0 else "Error %s/%s" % (element1, element2))

Comment (ldoktor) [inline io/disk/iozone.py:78]: Well we can handle that without the division, which would add an overhead and as this is a corner case, I'd suggest using

``` python
try:
    ratio = float(element2) / float(element1)
except ZeroDivisionError:
    new_line.append("." if float(element2) == 0 else "Error %s/%s" % (element1, element2))
    continue
```

File: io/disk/iozone.py (hunk 7 above)
Comment (harish-24) [inline io/disk/iozone.py:87]: Thanks for the suggestion.

File: io/disk/iozone.py (hunk 8 above)
Comment (harish-24) [inline io/disk/iozone.py:78]: Thanks.
